users/serializers.py

```diff
@@ -22,7 +24,11 @@ def create(self, validated_data):
         instance = self.Meta.model(**validated_data)  # as long as the fields are the same, we can just use this
         if password is not None:
             instance.set_password(password)
-        instance.save()
+        try:
+            instance.save()
+        except IntegrityError:
+            return HttpResponseBadRequest('User with given email already exist.')
```

Comment:
I still have error 500 during create user with already existing email in db. I found one solution with moving try-catch block one layer higher (serializer -> view) eg:

            try:
                user = serializer.save()
            except IntegrityError:
                return HttpResponseBadRequest('User with given email already exist.')

![image](https://user-images.githubusercontent.com/29740768/102516841-9c32ad80-408f-11eb-9318-8423c3977a3f.png)

Reply:
fixed